src/mutations/draft/putDraft.ts

```diff
@@ -113,46 +115,37 @@ const resolver: MutationToPutDraftResolver = async (
     coverId = asset.id
   }
 
-  // check for collection existence
-  // add to dbId array if ok
-  let collectionIds // leave as undefined // = null
-  if (collection) {
-    collectionIds = await Promise.all(
-      collection.map(async (articleGlobalId) => {
-        if (!articleGlobalId) {
-          return
-        }
-
-        const { id: articleId } = fromGlobalId(articleGlobalId)
-        const article = await articleService.baseFindById(articleId)
-
-        if (!article) {
-          throw new ArticleNotFoundError(
-            `Cannot find article ${articleGlobalId}`
-          )
-        }
-
-        if (article.state !== ARTICLE_STATE.active) {
-          throw new ForbiddenError(
-            `Article ${articleGlobalId} cannot be collected.`
-          )
-        }
-
-        const isBlocked = await userService.blocked({
-          userId: article.authorId,
-          targetId: viewer.id,
-        })
+  // handle Collection
+  const newCollectionIds =
+    collection == null
+      ? []
+      : _.uniq(
+          collection
+            .filter(_.isString)
```

Comment:
As the schema defined, `collection` is `[ID]`, an array of strings, so is this unnecessary?

Reply:
if `collection` is` [ID]` is `[ID!]`, it`s an array of strings.  Maybe the schema is wrong?

Reply:
Hmm, yes it should be `[ID!]`. Let's leave it here cuz any schema changes require client-side too.